weld/llvm.rs

```diff
@@ -38,6 +38,22 @@ static DICTIONARY_CODE: &'static str = include_str!("resources/dictionary.ll");
 static DICTMERGER_CODE: &'static str = include_str!("resources/dictmerger.ll");
 static GROUPMERGER_CODE: &'static str = include_str!("resources/groupbuilder.ll");
 
+/// Replace occurances of each `key` in `template` with `value`.
+///
+/// // returns "My name is Shoumik"
+/// replace!("My name is $NAME", "$NAME" => "shoumik")
```

Comment:
Wouldn't it be better to use `format!` for this? Maybe we'd have to change our templates a bit but it cloud be more consistent, especially since some .ll templates do use the `format!` format.